pyscf/pbc/scf/hf.py

```diff
@@ -237,27 +237,30 @@ def dip_moment(cell, dm, unit='Debye', verbose=logger.NOTE,
 
     if grids is None:
         grids = gen_grid.UniformGrids(cell)
-        #? FIXME: Less requirements on the density accuracy.
-        #ke_cutoff = gto.estimate_ke_cutoff(cell, 1e-5)
-        #grids.mesh = tools.cutoff_to_mesh(a, ke_cutoff)
+
     if rho is None:
         rho = numint.NumInt().get_rho(cell, dm, grids, kpt, cell.max_memory)
 
+    # With the optimal origin of the unti cell, the net dipole in the unit
+    # cell should be strictly zero. However, the integral grids are often not
+    # enough to produce the zero dipole. Errors are caused by the sub-optimal
+    # origin and the numerial integration.
```

Comment:
I think the dipole moment calculation should follow the implementation in [here](https://github.com/pyscf/properties/blob/92a4df7645033349232c4776355f8e53a9256987/pyscf/pbc/prop/polarizability/rhf.py#L237), although the accuracy of k-point derivatives need to be improved...

Reply:
The definition of dipole in the cell is vague. It depends on how the box of the cell is constructed. The dipole computed in this function is very likely different to the one you referred. It optimizes the cell center so as to minimize the dipole in the cell. This routine is used in this module to compute the Makov-Payne correction for charged systems. I'm not sure if this is the correct way to define dipole in a PBC system. Is the one you referred the better definition for dipole in pbc systems? If so could you please update the dipole function in a new PR?